From 2592c128be5547765c86d759551ce017cfb3d815 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:13 -0400
Subject: [powerpc] boot: Fix compile warning in 64bit

Message-id: <1410545655-205645-505-git-send-email-dzickus@redhat.com>
Patchwork-id: 94481
O-Subject: [RHEL7 PATCH 504/626] powerpc/boot: Fix compile warning in 64bit
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit b636031a7b0622ffdc5689ff36fb13669f00e160
Author: Cedric Le Goater <clg@fr.ibm.com>
Date:   Thu Apr 24 09:23:32 2014 +0200

    powerpc/boot: Fix compile warning in 64bit

    arch/powerpc/boot/oflib.c:211:9: warning: cast to pointer from integer of \
        different size [-Wint-to-pointer-cast]
      return (phandle) of_call_prom("finddevice", 1, 1, name);

    This is a work around. The definite solution would be to define the
    phandle typedef as a u32, as in the kernel, but this would break the
    device tree ops API.

    Let it be for the moment.

    Signed-off-by: Cedric Le Goater <clg@fr.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/boot/oflib.c b/arch/powerpc/boot/oflib.c
index 956443f..cdfe762 100644
--- a/arch/powerpc/boot/oflib.c
+++ b/arch/powerpc/boot/oflib.c
@@ -201,7 +201,7 @@ void of_exit(void)
  */
 void *of_finddevice(const char *name)
 {
- return (phandle) of_call_prom("finddevice", 1, 1, name);
+ return (void *) (unsigned long) of_call_prom("finddevice", 1, 1, name);
 }
 
 int of_getprop(const void *phandle, const char *name, void *buf,
-- 
1.7.1